pages/live.tsx

```diff
@@ -0,0 +1,203 @@
+              <>
+                Visit SSW TV Channel on Youtube
+                <BsArrowRightCircle className="ml-1 inline" />
+              </>
+            }
+            noAnimate
+            isTargetBlank={true}
+          />
+        </div>
+      </Container>
+      <Container size="xsmall">
+        <span className="text-sswRed">
+          <h2>{data.live.section2}</h2>
+        </span>
+        <div className="grid grid-cols-1 justify-center gap-8 lg:grid-cols-3">
+          {props.playListVideosLinks
+            .slice(0, visibleVideos)
+            .map((video, index) => (
+              <div key={index}>
+                <VideoCard {...video} theme="light" />
+              </div>
+            ))}
+        </div>
+        {props.playListVideosLinks.length > visibleVideos && (
+          <div className="flex justify-center">
+            <button
+              onClick={() => setVisibleVideos((prevCount) => prevCount + 9)}
+              className="m-8 text-sswRed underline"
+            >
+              See More
+            </button>
+          </div>
+        )}
+        <div className="flex justify-center">
+          <UtilityButton
+            size="small"
+            uncentered={false}
+            link={`https://www.youtube.com/playlist?list=${data.live.youtubePlaylistId}`}
+            buttonText={
+              <>
+                Watch more on SSW TV Channel on Youtube
+                <BsArrowRightCircle className="ml-1 inline" />
+              </>
+            }
+            noAnimate
+            isTargetBlank={true}
+          />
+        </div>
+      </Container>
+      <BuiltOnAzure data={{ backgroundColor: "lightgray" }} />
+    </Layout>
+  );
+}
+
+export const getStaticProps = async () => {
+  const tinaProps = await client.queries.liveContentQuery({
+    relativePath: "index.mdx",
+  });
+  const res = await getYoutubePlaylist(tinaProps.data.live.youtubePlaylistId);
+
+  let playListVideosLinks = [];
+
+  if (res && res.items) {
+    const playListVideosBaseUrl = "https://www.youtube.com/watch?v=";
+
+    playListVideosLinks = res.items.map((item) => {
+      const snippet = item.snippet;
+      const title = snippet.title;
+      const playListId = snippet.playlistId;
+      const videoId = snippet.resourceId.videoId;
+
+      return {
+        title: title,
+        link: `${playListVideosBaseUrl}${videoId}&list=${playListId}`,
+      };
+    });
+  }
+
+  const currentDate = new Date().toISOString();
+
+  const events = await getEvents(
+    `$filter=fields/Enabled ne false and fields/EndDateTime gt '${currentDate}' and fields/CalendarType eq 'User Groups'&$orderby=fields/StartDateTime asc`
+  );
+
+  let event = events[0];
+
+  if (!event) {
+    const pastEvents = await getEvents(
+      `$filter=fields/Enabled ne false and fields/EndDateTime lt '${currentDate}' and fields/CalendarType eq 'User Groups'&$orderby=fields/StartDateTime desc`
+    );
+
+    event = pastEvents[0];
+  }
```

Comment:
Can you abstract this to a file in `services/server` as its the same as in `pages/netug/[[..filename]].tsx`

Reply:
Done - it is located in services | server | events.ts